controllers/awsmachine_controller.go

```diff
@@ -174,7 +174,7 @@ func (r *AWSMachineReconciler) SetupWithManager(mgr ctrl.Manager, options contro
 func (r *AWSMachineReconciler) reconcileDelete(machineScope *scope.MachineScope, clusterScope *scope.ClusterScope) (reconcile.Result, error) {
 	machineScope.Info("Handling deleted AWSMachine")
 
-	ec2Service := r.serviceFactory(clusterScope)
+	ec2Service := r.getEC2Service(clusterScope)
```

Comment:
I think `serviceFactory` is for unit testing (see https://github.com/kubernetes-sigs/cluster-api-provider-aws/commit/101c68eac91844ad01ba366b5f43f072888525a3). We probably need to default the variable in the constructor so it works for the normal path (non units)

Reply:
Nevermind, read the code fully.